lib/utmp.c

```diff
@@ -157,7 +157,8 @@ get_current_utmp(void)
 
 	/* First, try to find a valid utmp entry for this process.  */
 	while ((ut = getutxent()) != NULL) {
-		if (   (ut->ut_pid == getpid ())
+		if (   (   (ut->ut_pid == getpid ())
+		        || (ut->ut_pid == getppid ()))
```

Comment:
(sorry I'm late, but)

Is there a chance that in fact both pid and ppid have (different) valid entries already?  Would it be better to cache the ut if ut->ut->pid == getppid(), keep running through the list, and then only return the cached ut if no ut->ut_pid == getpid() was found?

Reply:
@Karlson2k

Reply:
My reply is here: https://github.com/shadow-maint/shadow/pull/1282#issuecomment-3068704121